pkg/build/users_test.go

```diff
@@ -0,0 +1,104 @@
+package build
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/suse-edge/edge-image-builder/pkg/config"
+)
+
+func TestConfigureUsers(t *testing.T) {
+	// Setup
+	configData, err := os.ReadFile("./testdata/users-test-definition.yaml")
+	require.NoError(t, err)
+	imageConfig, err := config.Parse(configData)
```

Comment:
I'd suggest to drop parsing from these tests. What we want to verify here is that `configureUsers` is working as expected. Manually populating the config should be more than enough to cover these needs and decouple the config implementation from the resulting script assertions:

```
imageConfig = &config.ImageConfig{
   OperatingSystem: config.OperatingSystem{
	Users: []config.OperatingSystemUser{
		{
			Username: "alpha",
			Password: "alpha123",
			SSHKey:   "alphakey",
		},
		...
	},
   },
}
```

Note that I fully agree that we should have "integration" tests down the line involving fully fleshed YAML configs, however, here we're just testing the users unit.

Reply:
This makes sense and it'll be nice to remove the extra files. In hindsight, I took this approach because I wasn't sure how the parser would handle no OS section in the other test (would it be nil or still created) and then never rethought the approach.